ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/deduplication/HexOffsetSpec.scala

```diff
@@ -45,15 +47,19 @@ class HexOffsetSpec
           "00007a94" -> "00007a93",
         )
       ) { case (offset, expectedOffset) =>
-        HexOffset.firstBefore(
+        HexOffset.previous(
           Ref.HexString.assertFromString(offset)
         ) shouldBe Some(Ref.HexString.assertFromString(expectedOffset))
       }
     }
 
     "return lower offset" in forAll(offsets) { offset =>
-      val lowerOffset = HexOffset.firstBefore(offset).value
-      offset should be > [String] lowerOffset
+      if (new BigInteger(offset, 16) == BigInteger.ZERO) {
```

Comment:
Just to make sure, was it flaky?

Reply:
Yes and no. 
No in the sense that it never failed so far.
Yes because eventually, the generator would generate a hex which would be composed only from 0s. 
The probability would be quite low, as the min length is 2 and that would mean choosing '0' from '[a-f|0-9]' twice in a row, but it would've eventually have happened.